CHANGELOG.md

```diff
@@ -1,3 +1,7 @@
+## 0.12.6
+
+- Add `isAssertionError` type matcher.
```

Comment:
Do we want a `throwAssertionError` to go with it?

Reply:
I didn't see other `throwsA...` definitions in the matcher package. Would you prefer I add this to the same file or elsewhere? Or we can leave it as is and align with the behavior that other errors have (i.e. the user would have to type `throwsA(const TypeMatcher<AssertionError>())`)?

Another option is to add `throwsA...` definitions for all the errors defined in error_matcher.dart. I'm also happy to do that.

Reply:
Ah my fault - they are defined in package:test_api not here.
https://github.com/dart-lang/test/blob/master/pkgs/test_api/lib/src/frontend/throws_matchers.dart

Reply:
Okay, then I'll leave this PR as is.